packages/stream_video/lib/src/stream_video.dart

```diff
@@ -251,6 +251,10 @@ class StreamVideo {
 
   /// Connects the user to the Stream Video service.
   Future<Result<UserToken>> connect() async {
+    if (currentUserType == UserType.anonymous) {
+      _logger.w(() => '[connect] rejected (anonymous user)');
+      return Result.error('Cannot connect anonymous user to the WS');
```

Comment:
Should this be considered an erorr, @kanat?

For example, if I am following the tutorial and decide to connect as a guest, an error here might be a deceiving.

Reply:
There are 3 user types:
- authenticated
- guest
- anonymous

For `anonymous` we should not connect to the WS.

This logic is common for all SDKs.
Here is the iOS [implementation](https://github.com/GetStream/stream-video-swift/blob/main/Sources/StreamVideo/StreamVideo.swift#L503-L507)

Reply:
That's fair. Can we update the text there to be a bit more descriptive? 

iOS for example uses "Missing Permissions" as the error message.

Reply:
Updated 👍